components/Table/Table.js

```diff
@@ -95,21 +152,139 @@ export class Table {
+
+    const filterLabel = document.createElement('div');
+    filterLabel.classList.add('filter-label');
+    filterLabel.textContent = 'Name';
+
+    const filterOperator = document.createElement('div');
+    filterOperator.classList.add('filter-operator');
+    filterOperator.textContent = 'equal';
+
+    const filterValue = document.createElement('input');
+    filterValue.classList.add('filter-value');
+    filterValue.placeholder = 'Filter value';
+
+    const btnClean = document.createElement('div');
+    btnClean.classList.add('btn-clean');
+    btnClean.textContent = 'x';
+
+    const btnExit = document.createElement('div');
+    btnExit.classList.add('btn-exit');
+    btnExit.textContent = 'x';
+
+    btnExit.addEventListener('click', () => {
+      filterMenuContainer.parentElement.children[1].remove();
+    });
+
+    btnClean.addEventListener('click', () => {
+      filterValue.value = '';
+      document
+        .querySelectorAll('.table-row')
+        .forEach((row) => (row.style.display = 'flex'));
+    });
+
+    const filterFunction = (e) => {
+      let filter = e.target.value;
+      this.callbacksMap.filterCallback(filter);
+    };
+
+    filterValue.addEventListener('input', filterFunction);
+
+    filterMenuContainer.append(
+      btnClean,
+      filterLabel,
+      filterOperator,
+      filterValue,
+      btnExit
+    );
+    this.tableHeader.append(filterMenuContainer);
+  }
+
+  createPagination(TOTAL_AMOUNT = 0) {
+    const pagintaionContainer = document.createElement('div');
+    pagintaionContainer.classList.add('pagination-container');
+
+    const dropdownPagintaionContainer = document.createElement('select');
+    dropdownPagintaionContainer.classList.add('dropdown-pagination-container');
+
+    const pageNumberContainer = document.createElement('ul');
+    pageNumberContainer.classList.add('page-num-container');
+
+    const renderPagesNumbers = () => {
+      const pages = Math.ceil(TOTAL_AMOUNT / this.amount);
+
+      for (let i = 1; i <= pages; i++) {
+        const pageNumberBtn = document.createElement('li');
+        pageNumberBtn.classList.add('page-number');
+        pageNumberBtn.textContent = i;
+        pageNumberContainer.append(pageNumberBtn);
+
+        pageNumberBtn.addEventListener('click', () => {
+          const currentPage = +pageNumberBtn.innerHTML;
+          this.callbacksMap.paginationCallback(currentPage, this.amount);
+        });
+      }
+    };
+    renderPagesNumbers();
+
+    const dropdownPagintaionItems = this.PaginationConfigList.map((item) => {
+      const dropPagItem = document.createElement('option');
+      dropPagItem.classList.add('drop-pag-item');
+      dropPagItem.textContent = item.pageSize;
+
+      dropdownPagintaionContainer.addEventListener('change', (e) => {
+        if (this.amount !== +e.target.value) {
+          pageNumberContainer.innerHTML = '';
+          this.amount = +e.target.value;
+          renderPagesNumbers();
```

Comment:
You should decide to rebuild your pagination buttons when you get the amount of the results
As I see here, you're rebuilding whole the pagination when you call this method
Why do we need it?
We can leave our select for the amount as is
Only one thing, that we should rebuild based on the amount - is our amount of the pages